include/triton/Dialect/Triton/IR/TritonAttrDefs.td

```diff
@@ -119,15 +119,16 @@ def TT_InputPrecisionAttr : I32EnumAttr<
   let cppNamespace = "::mlir::triton";
 }
 
-// Type for F8F6F4 kind of floats.
-def TT_F8F6F4TypeAttr : I32EnumAttr<
-    "F8F6F4Type", "",
+// Type for ScaleType kind of floats.
+def TT_ScaleTypeAttr : I32EnumAttr<
```

Comment:
You mean that the scale is an I8 if represents an fp4? That's fair. Any suggestion of a better name?

Reply:
Also, note that a better design for the TTIR would be to pass in Optional<SubByteType>, and have the convention that either the scale has the correct floating type, or it's an U8, and subByteType is not None).
I didn't do this because handling Optionals in MLIR, in particular when they are consecutive, is a bit of a pain.
I'm happy to rename it, or implement this other approach at a later stage (or now if you really think we should do it).

Reply:
> Any suggestion of a better name?

In the microscaling spec they call this part element data type, so maybe `ElementTypeAttr` or `ElementDataTypeAttr`, I don't like it a lot though so if you find something better that's fine with me. (scale made it particularly confusing to me)

> Also, note that a better design for the TTIR would be to pass in Optional, and have the convention that either the scale has the correct floating type, or it's an U8, and subByteType is not None).
I didn't do this because handling Optionals in MLIR, in particular when they are consecutive, is a bit of a pain.
I'm happy to rename it, or implement this other approach at a later stage (or now if you really think we should do it).

Making it optional in the language would be nice but in the IR I would think it is better to have redundant information rather than multiples way to represent it.

Reply:
went with `ScaleDotElemType`. We can bikeshed it at a later point.

Reply:
bad rebase, should be ready now